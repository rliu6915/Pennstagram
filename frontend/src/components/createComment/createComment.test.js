/* eslint-disable no-undef */
// import testing library functions
import '@testing-library/jest-dom/extend-expect';
// eslint-disable-next-line no-unused-vars
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CreateComment from './createComment';

// Snapshot testing

test('createCommentComponent', () => {
  const component = renderer.create(<CreateComment />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
/*
// unit test for the react component
// test block
test('Create Comment', () => {
  // render the component on virtual dom
  render(<CreateComment />);
  // select the elements
  const wrapper = document.getElementsByClassName('createCommentWrapper');
  expect(wrapper).toBeInTheDocument();
  const btn = document.getElementsByClassName('commentSubmit');
  expect(btn).toBeInTheDocument();
});
*/
// describe("render commentCreated", () => {
//   test('renders a commentCreated page', async() => {
//     const { container } = render(
//       <CreateComment />
//     );
//     container.hasAttribute('commentCreated');
// })
// })
