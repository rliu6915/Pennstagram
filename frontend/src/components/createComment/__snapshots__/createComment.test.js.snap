// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCommentComponent 1`] = `
<div>
  <section
    className="createCommentWrapper"
  >
    <div
      className="createcontrol"
    >
      <div
        style={
          Object {
            "overflowY": "visible",
            "position": "relative",
          }
        }
      >
        <div
          style={Object {}}
        >
          <div
            style={
              Object {
                "border": "1px solid transparent",
                "boxSizing": "border-box",
                "color": "transparent",
                "overflow": "hidden",
                "position": "relative",
                "textAlign": "start",
                "whiteSpace": "pre-wrap",
                "width": "100%",
                "wordWrap": "break-word",
              }
            }
          >
             
          </div>
          <textarea
            id="commentContent"
            onBlur={[Function]}
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onScroll={[Function]}
            onSelect={[Function]}
            placeholder="Leave a comment or mention people using '@'"
            style={
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "fontFamily": "inherit",
                "fontSize": "inherit",
                "height": "100%",
                "left": 0,
                "letterSpacing": "inherit",
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "resize": "none",
                "top": 0,
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="commentSubmit"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </section>
</div>
`;
